src/routers/datatypeRouters/nodes/proteins.ts

```diff
@@ -82,6 +82,18 @@ async function findProteinsByTextSearch (input: paramsFormatType): Promise<any[]
     delete input.limit
   }
 
+  const exactQuery = `
+    FOR record IN ${proteinSchema.db_collection_name}
+    FILTER ${getFilterStatements(proteinSchema, input)}
+    LIMIT ${input.page as number * limit}, ${limit}
+    RETURN { ${getDBReturnStatements(proteinSchema)} }
+  `
+
+  const exactObjects = await (await db.query(exactQuery)).all()
+  if (exactObjects.length !== 0) {
+    return exactObjects
+  }
+
```

Comment:
I feel this part is redundant compare to function findProteins. Can you refactor to remove the redundancy?

Reply:
just pushed a refactoring